module/zfs/vdev_disk.c

```diff
@@ -593,32 +583,20 @@ __vdev_disk_physio(struct block_device *bdev, zio_t *zio, caddr_t kbuf_ptr,
 		if (dr->dr_bio[i])
 			vdev_submit_bio(rw, dr->dr_bio[i]);
 
-	/*
-	 * On synchronous blocking requests we wait for all bio the completion
-	 * callbacks to run.  We will be woken when the last callback runs
-	 * for this dio.  We are responsible for putting the last dio_request
-	 * reference will in turn put back the last bio references.  The
-	 * only synchronous consumer is vdev_disk_read_rootlabel() all other
-	 * IO originating from vdev_disk_io_start() is asynchronous.
-	 */
-	if (wait) {
-		wait_for_completion(&dr->dr_comp);
-		error = dr->dr_error;
-		ASSERT3S(atomic_read(&dr->dr_ref), ==, 1);
-	}
-
 	(void) vdev_disk_dio_put(dr);
 
 	return (error);
 }
 
+#ifndef __linux__
```

Comment:
This whole file is already heavily Linux specific so we could simply remove these functions entirely.  However, I don't object to keeping them around and disabled like this in case we determine we need them.